docs/tutorial/beginning_decomp.md

```diff
@@ -0,0 +1,673 @@
+            phi_s0 = temp_s0;
+        } while (temp_s0 != D_80C106C0);
+        D_80C106C8 = 1;
+    }
+    this->unk_2AC = 2;
+    if (Flags_GetSwitch(globalCtx, (s32) this->actor.params) != 0) {
+        this->actor.textId = 0x2ADC;
+    } else {
+        this->actor.textId = 0x2AD9;
+    }
+    func_80C10148(this);
+}
+```
+Comment out the `GLOBAL_ASM` line for `Init`, and paste all of this into the file just underneath it:
+
+```C
+[...]
+// #pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Recepgirl/EnRecepgirl_Init.s")
+? func_80C10148(EnRecepgirl *); // extern
+extern FlexSkeletonHeader D_06011B60;
+static void *D_80C106B0[4] = {(void *)0x600F8F0, (void *)0x600FCF0, (void *)0x60100F0, (void *)0x600FCF0};
+static s32 D_80C106C8 = 0;
+InitChainEntry D_80C106C0[2]; // unable to generate initializer
+
+void EnRecepgirl_Init(EnRecepgirl *this, GlobalContext *globalCtx) {
+    void **temp_s0;
+    void **phi_s0;
+
+    Actor_ProcessInitChain((Actor *) this, D_80C106C0);
+    ActorShape_Init(&this->actor.shape, -60.0f, NULL, 0.0f);
+    SkelAnime_InitFlex(globalCtx, (SkelAnime *) this->unk_144, &D_06011B60, (AnimationHeader *) &D_06009890, this + 0x188, this + 0x218, 0x18);
+    phi_s0 = D_80C106B0;
+    if (D_80C106C8 == 0) {
+        do {
+            temp_s0 = phi_s0 + 4;
+            temp_s0->unk-4 = Lib_SegmentedToVirtual(*phi_s0);
+            phi_s0 = temp_s0;
+        } while (temp_s0 != D_80C106C0);
+        D_80C106C8 = 1;
+    }
+    this->unk_2AC = 2;
+    if (Flags_GetSwitch(globalCtx, (s32) this->actor.params) != 0) {
+        this->actor.textId = 0x2ADC;
+    } else {
+        this->actor.textId = 0x2AD9;
+    }
+    func_80C10148(this);
+}
+[...]
+```
+</details>
+
+Typically for all but the simplest functions, there is a lot that needs fixing before we are anywhere near seeing how close we are to the original code. You will notice that mips2c creates a lot of temporary variables. Usually most of these will turn out to not be real, and we need to remove the right ones to get the code to match.
+
+First, change the first argument back to `Actor* thisx` so that the function matches its prototype above. To allow the function to find the variables, we need another correction. Half of this has already been done at the top of the function, where we have
+
+```C
+#define THIS ((EnRecepgirl*)thisx)
+```
+
+To do the other half, write the following at the beginning of the function, before any declarations:
+
+```C
+EnRecepgirl* this = THIS;
+```
+
+Now everything points to the right place, even though the argument of the function seems inconsistent with the contents.
+
+(Again: this step is only necessary for the "main four" functions, and sometimes functions that are used by these: it relates to how such functions are used outside the actor.)
+
+While we are carrying out initial changes, you can also find-and-replace any instances of `(Actor *) this` by `&this->actor`. The function now looks like this:
```

Comment:
This section could also be edited to reflect the other change, sorry I didn't mark it before.
```suggestion
To allow the function to find the variables, we need another correction. Half of this has already been done at the top of the function, where we have

``C
#define THIS ((EnRecepgirl*)thisx)
``

To do the other half, use this macro to replace the declaration of `this` at the beginning of the function:

``C
EnRecepgirl* this = THIS;
``

While we are carrying out initial changes, you can also find-and-replace any instances of `(Actor *) this` by `&this->actor`. The function now looks like this:
```

This last bit *"you can also find-and-replace any instances of `(Actor *) this`"*, is probably not strictly necessary, but I don't think it hurts to leave in.

(NB: I can't use triple-backticks in a suggestion, so you'll have to make this edit yourself instead of using the "accept suggestion" feature)

Reply:
Is this section necessary at all now?

Reply:
I don't think it's critical; I the only thing is that `mips2c` will output something like:
```c
EnRecepgirl* this = (EnRecepgirl *) thisx;
```
which can be rewritten to use `THIS`.

Reply:
Ah right, yeah that probably is worth adding.